### DISCUSSION TITLE
What is data-wails-* for?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [wailsapp/wails#854] [Q&A]
I tried to create a super simple FORM with an onsubmit function. But my function code was never called.

I created a plain vanilla project, where the greet button works. I added my form, nothing, same as my code. I removed the `data-wails-no-drag` tag (?) and the vanilla code didn't work too.

So, it seems to get any interaction up & running I somehow need to use this tag (?).
* What is it?
* Is there a documentation?
* When to use it?
* Where to use it?
* Are there are other tags I should be aware about?

**Comment by leaanthony** [comment]
The docs for it are here: https://wails.io/docs/guides/frameless
It's unrelated to javascript or form processing.

I've just realised search in the docs aren't enabled. I'll fix that.

**Reply by Robert-M-Muench** [reply]
Aha... ok. And the side effect is that buttons are becoming drag-able and the clicks don't pass through.

That's a bit of strange behavior because the visual feedback is a clear button click, and the window is not moving. Hence, it's perplexing to make the link to the frameless topic.

I ran into this because the default projects, which I use as sample, use frameless mode. Maybe better to really use a super stripped down standard window, with no hidden side effects.

**Reply by leaanthony** [reply]
Why are buttons becoming draggable? You can control this by turning drag off for the element. The page I linked to has the details

**Reply by Robert-M-Muench** [reply]
To summarize the problem, which either is a bug or a subtle thing, one has to know:

If a button is within a section that is marked with `data-wails-drag` the button is no longer firing click events.

**Reply by leaanthony** [reply]
That's correct and not a bug. If you want a button to be clickable, it shouldn't be a drag handle. I think passing through the event as well as dragging the window would be confusing. What do you think?

**Reply by Robert-M-Muench** [reply]
I would default to implicitly make all clickable things, `no-drag`. It doesn't make sense to use these for dragging. I'm sure most people forget to add the `no-drag` (if the rest is draggable) in the first round.

However, IIRC I removed all the tags and the problem persisted like no tag implicitly become `data-wails-drag` but I might be wrong.

**Reply by leaanthony** [reply]
I think this would be too hard. You can add `onClick` to any element. I suspect the drag handle use case to be limited to a section of the page without clickable elements.

**Comment by Robert-M-Muench** [comment]
Even more strange is, that I use `Frameless:         false,` in main.go

Shouldn't this avoid any frameless side effects?

**Reply by leaanthony** [reply]
There's no side effect magic. If you want an element to be draggable, add the attribute. Whether it's frameless or not doesn't matter.

**Reply by Robert-M-Muench** [reply]
I was confused because of the link to the frameless page. Gave me a wrong context.

**Reply by leaanthony** [reply]
That's fair. Where would you expect that information to be?

**Reply by Robert-M-Muench** [reply]
I would split the frameless and draggable stuff into two separate pages. Give it an own page, as it's a topic of its own.

**Reply by leaanthony** [reply]
There wouldn't really be a frameless page though without draggable. I'll have a think about it
